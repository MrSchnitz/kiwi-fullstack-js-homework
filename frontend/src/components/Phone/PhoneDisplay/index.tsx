import React from "react";
import { Textarea } from "@kiwicom/orbit-components/lib";
import styled, {keyframes} from "styled-components";

interface PhoneDisplayProps {
  text: string;
  onChange?: (value: string) => void;
}

const PhoneDisplay: React.FC<PhoneDisplayProps> = ({
  text,
  onChange,
}: PhoneDisplayProps) => {
  return (
    <TextareaLayout>
      <Textarea
        size={"normal"}
        readOnly={true}
        fullHeight={true}
        placeholder="Type something..."
        value={text.length > 0 ? text : "Type something..."}
      />
    </TextareaLayout>
  );
};

const TextareaLayoutAnimation = keyframes`
    0% {
        opacity: 0;
    }
    100% {
        opacity: 1;
    }
`;

const TextareaLayout = styled.div`
  height: 100%;
  width: 100%;
  font-size: 3rem;
  
  animation: ${TextareaLayoutAnimation} 4s ease;

  textarea {
    font-size: 2.5rem;
    padding: 1.5rem;
    border: none;
    outline: none;
    border-color: transparent;
    background-color: #d6ead9;
    // background-color: #ECF8F7;
  }
`;

export default PhoneDisplay;
