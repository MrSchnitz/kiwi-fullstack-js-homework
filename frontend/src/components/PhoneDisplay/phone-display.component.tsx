import React from "react";
import {Textarea} from "@kiwicom/orbit-components/lib";
import {TextareaLayout} from "./phone-display.styles";

interface PhoneDisplayProps {
  text: string;
  onChange?: (value: string) => void;
}

const PhoneDisplay: React.FC<PhoneDisplayProps> = ({
  text,
  onChange,
}: PhoneDisplayProps) => {
  return (
    <TextareaLayout>
      <Textarea
        size={"normal"}
        readOnly={true}
        fullHeight={true}
        placeholder="Type something..."
        value={text.length > 0 ? text : "Type something..."}
      />
    </TextareaLayout>
  );
};

export default PhoneDisplay;
