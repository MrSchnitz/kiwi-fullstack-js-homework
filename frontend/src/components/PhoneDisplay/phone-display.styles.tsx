import styled, { keyframes } from "styled-components";

const TextareaLayoutAnimation = keyframes`
    0% {
        opacity: 0;
    }
    100% {
        opacity: 1;
    }
`;

export const TextareaLayout = styled.div`
  height: 100%;
  width: 100%;
  font-size: 3rem;

  animation: ${TextareaLayoutAnimation} 4s ease;

  textarea {
    font-size: 2.5rem;
    padding: 1.5rem;
    border: none;
    outline: none;
    border-color: transparent;
    background-color: #d6ead9;
  }
`;
