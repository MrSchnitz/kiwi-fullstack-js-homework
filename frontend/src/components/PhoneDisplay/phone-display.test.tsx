import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import PhoneDisplay, { PhoneDisplayProps } from "./phone-display.component";

describe("PhoneDisplay component", () => {
  let wrapper: ShallowWrapper;

  const mockProps: PhoneDisplayProps = {
    text: "home",
  };

  it("should render PhoneDisplay component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should display text", () => {
    wrapper = shallow(<PhoneDisplay {...mockProps} />);
    expect(wrapper.find("Textarea").prop("value")).toBe(mockProps.text);
  });
});
