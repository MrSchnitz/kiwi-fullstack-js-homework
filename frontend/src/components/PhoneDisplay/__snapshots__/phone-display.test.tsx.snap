// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneDisplay component should render PhoneDisplay component 1`] = `undefined`;
